R/shared-methods.R

```diff
@@ -149,6 +151,9 @@ DataClass <- R6::R6Class(
     #' @field target_regions A character vector of regions to filter for. Used
     #' by the `filter method`.
     target_regions = NULL,
+    #' @field process_fns array, additional, user supplied functions to process
+    #' the data.
+    process_fns = c(),
```

Comment:
maybe give this a default that is also changed at initialisation if supplied to help with docs?

Reply:
doesn't make much difference

Reply:
I'd rather leave it empty as if its populated it could cause confusion as to why its being overwriten at initialisation

Reply:
The logic here is: 

```
if (!missing(process_fns)) {
self$process_fns <- process_fns
}
```

Rather than. overwriting.

Reply:
I see, I thought we would have to set it as a default argument then overwrite `self$process_fns` but this is clean.

Reply:
its a bit 50/50 but this maybe makes it easier/nicer to doc.